### PR TITLE
Nested private registry lookup support

### PR DESCRIPTION
**Pull request by alena1108**
https://github.com/rancher/rke/issues/1232

### CODE REVIEW

File: docker/docker.go [hunk 1]
```diff
@@ -634,16 +638,53 @@ func getRegistryAuth(pr v3.PrivateRegistry) (string, error) {
 }
 
 func GetImageRegistryConfig(image string, prsMap map[string]v3.PrivateRegistry) (string, string, error) {
+	/*
+		Image can be passed as
+		- repo.com/foo/bar/rancher/rke-tools:v0.1.51
```
Comment (leodotcloud) [inline docker/docker.go]: Does adding a test make sense to test this function?

File: docker/docker.go [hunk 2]
```diff
@@ -634,16 +638,53 @@ func getRegistryAuth(pr v3.PrivateRegistry) (string, error) {
...
+	if len(prsMap) == 0 {
+		return "", "", nil
+	}
 	regURL := ref.Domain(namedImage)
-	if pr, ok := prsMap[regURL]; ok {
-		// We do this if we have some docker.io login information
-		regAuth, err := getRegistryAuth(pr)
-		return regAuth, pr.URL, err
+	regPath := ref.Path(namedImage)
+	splitPath := strings.Split(regPath, "/")
```
Comment (leodotcloud) [inline docker/docker.go]: If the user has provided `repo.com/foo/bar` as the registry URL, why do we have to check repo.com/foo and repo.com? That seems like an unintended side/extra behavior and we would have to document this behavior if we want to provide this functionality. If the user intends to use these different search paths, it's better to have the same explicitly defined in the config.

Comment (alena1108) [reply docker/docker.go]: @leodotcloud the logic works this way:

* system images can be coming from diff repos (although rarely used, we've always had support for it in rke)
* For every image, we do the repo lookup. If we are to add search path explicitly, you'd have to do it for each and every image, and that's more cumbersome in my opinion.

File: docker/docker.go (hunk 1 above)
Comment (alena1108) [reply docker/docker.go]: @leodotcloud I can try adding a unittest. Don't think adding integration test would be possible with our current framework as we don't have long standing (and maintained) private repo

**Pull request merged**